posthog/ai/utils.py

```diff
@@ -29,6 +29,35 @@ def get_model_params(kwargs: Dict[str, Any]) -> Dict[str, Any]:
     return model_params
 
 
+def extract_core_model_params(kwargs: Dict[str, Any], provider: str) -> Dict[str, Any]:
+    """
+    Extracts core model parameters from the kwargs dictionary.
+    """
+    output = {}
+    if provider == "anthropic":
```

Comment:
To track AI cost processing's case-insensitive behavior:
```suggestion
	if provider:
	    provider = provider.lower()
    if provider == "anthropic":
```